http-server/src/main/java/com/proofpoint/http/server/HttpServerModule.java

```diff
@@ -56,11 +57,17 @@
 {
     public static final String REALM_NAME = "Proofpoint";
 
+    private SessionHandler sessionHandler;
+
     @Override
     public void configure(Binder binder)
     {
         binder.disableCircularProxies();
 
+        if (sessionHandler != null) {
+            binder.bind(SessionHandler.class).toInstance(sessionHandler);
+        }
```

Comment:
Please avoid unnecessary use of instance bindings. See https://github.com/google/guice/wiki/InstanceBindings

```suggestion
            if (bindSessionHandler) {
                binder.bind(SessionHandler.class).in(Scopes.SINGLETON)
            }
```